src/torcontrol.cpp

```diff
@@ -481,6 +483,53 @@ TorController::~TorController()
     }
 }
 
+void TorController::get_socks_cb(TorControlConnection& _conn, const TorControlReply& reply)
+{
+    // NOTE: We can only get here if -onion is unset
+    std::string socks_port_str;
+    if (reply.code == 250) {
+        for (const auto& line : reply.lines) {
+            if (0 == line.compare(0, 20, "net/listeners/socks=")) {
+                const std::string port_list_str = line.substr(20);
+                std::vector<std::string> port_list;
+                boost::split(port_list, port_list_str, boost::is_any_of(" "));
+                for (auto& portstr : port_list) {
```

Comment:
```suggestion
                for (const auto& portstr : spanparsing::Split(port_list_str, ' ')) {
```

Reply:
```
torcontrol.cpp:340:80: error: no member named 'rbegin' in 'Span<const char>'; did you mean 'begin'?
                    if ((portstr[0] == '"' || portstr[0] == '\'') && (*portstr.rbegin() == '"' || *portstr.rbegin() == '\'')) {
                                                                               ^~~~~~
```

Reply:
Ah, `Span` has `begin()`, but no `rbegin()`. It has `back()` to lookup the last element, so `*portstr.rbegin()` can be replaced `portstr.back()` if you like it better.